### ISSUE TITLE
The keys in the options argument to ReferenceTracker should be optional

### ISSUE THREAD

**Issue by MageJohn** [typescript-eslint/typescript-eslint#3526]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```TS
import { ASTUtils } from "@typescript-eslint/experimental-utils";
const { ReferenceTracker } = ASTUtils;

const tracker1 = new ReferenceTracker(context.getScope(), {});
const tracker2 = new ReferenceTracker(context.getScope(), {
  mode: "legacy"
});
const tracker3 = new ReferenceTracker(context.getScope(), {
  globalObjectNames: ["global", "globalThis", "self", "window"]
});
```

**Expected Result**

All the ReferenceTracker constructions above are should be valid, with no type errors.

**Actual Result**

Each of them produces a type error. For example, `tracker1` produces this:
```
Argument of type '{}' is not assignable to parameter of type '{ mode: "strict" | "legacy"; globalObjectNames: readonly string[]; }'.
  Type '{}' is missing the following properties from type '{ mode: "strict" | "legacy"; globalObjectNames: readonly string[]; }': mode, globalObjectNames
```

**Additional Info**

The options object for ReferenceTracker should have all it's keys as optional. The [implementation](https://github.com/mysticatea/eslint-utils/blob/c3b1d38328dc3a5278b208a58186a0735315f58f/src/reference-tracker.js#L63-L74) from eslint-utils looks like this:
```JS
constructor(
    globalScope,
    {
        mode = "strict",
        globalObjectNames = ["global", "globalThis", "self", "window"],
    } = {},
) {
  // ...
}
```

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `4.26.1` |
| `TypeScript`                            | `4.3.2` |
| `node`                                  | `12.22.0` |

**Comment by bradzacher** [comment]
Happy to accept a PR.
With this sort of thing it's usually better and faster to just raise a PR with context in the description rather than raising an issue first.

**Comment by MageJohn** [comment]
Cool, on it now!
